File: 백준/Gold/2631. 줄세우기/줄세우기.java
import java.io.*;
import java.util.*;


public class Main {
    static int n;
    static int [] dp = new int [201];
    static int [] arr = new int [201];
    static int maxdp = 0;

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine();
        n = Integer.parseInt(s);


        for (int i = 0; i <n ; i++) {
            arr[i] = Integer.parseInt(br.readLine());
            dp[i]=1;
        }


        for (int i = 0; i <n ; i++) {
            for (int j = 0; j <i; j++) {
                if(arr[i] > arr[j]){
                    dp[i] = Math.max(dp[i],dp[j]+1);
                    maxdp = Math.max(maxdp,dp[i]);
                }
            }
        }


        System.out.println(n-maxdp);


    }
}

File: 백준/Gold/2631. 줄세우기/줄세우기.cc
#include<bits/stdc++.h>
using namespace std;

int n;
int dp[201];
int arr[201];



int main() {

    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);


    cin >> n;

    for(int i = 0 ; i < n; i++){
        cin >> arr[i];
        dp[i]=1;
    }



    int maxdp=0;

    for(int i = 0 ; i < n; i++){
        for(int j = 0; j < i; j++){
            if(arr[i] > arr[j]){
                dp[i] = max(dp[i],dp[j]+1);
                maxdp = max(dp[i],maxdp);
            }
        }
    }
    cout << n-maxdp <<"\n";










    return 0;
}